src/site/twitch.tv/modules/media-player/MediaPlayer.vue

```diff
@@ -0,0 +1,155 @@
+<template>
+	<template v-if="shouldShowVideoStats">
+		<Teleport v-if="doTeleport" :to="teleportLocation">
+			<div
+				id="seventv-video-stats"
+				ref="videoStatsRef"
+				@mouseenter="videoStatsTooltip.show(videoStatsRef)"
+				@mouseleave="videoStatsTooltip.hide()"
+			>
+				<figure v-if="videoStats.playbackRate > 1"><ForwardIcon /></figure>
+				<figure v-else><GaugeIcon /></figure>
+				<span>{{ liveLatency }}s</span>
+			</div>
+		</Teleport>
+	</template>
+</template>
+
+<script setup lang="ts">
+import { onUnmounted, ref, toRef, watch } from "vue";
+import { HookedInstance } from "@/common/ReactHooks";
+import {
+	defineNamedEventHandler,
+	definePropertyHook,
+	unsetNamedEventHandler,
+	unsetPropertyHook,
+} from "@/common/Reflection";
+import { useConfig } from "@/composable/useSettings";
+import { useTooltip } from "@/composable/useTooltip";
+import ForwardIcon from "@/assets/svg/icons/ForwardIcon.vue";
+import GaugeIcon from "@/assets/svg/icons/GaugeIcon.vue";
+import VideoStatsTooltip from "./VideoStatsTooltip.vue";
+
+const props = defineProps<{
+	instance: HookedInstance<Twitch.MediaPlayerComponent>;
+	videoStatsTeleportLocation: HTMLElement | null | undefined;
+}>();
+
+const teleportLocation = toRef(props, "videoStatsTeleportLocation");
+const doTeleport = ref<boolean>(false);
+
+const shouldShowVideoStats = useConfig<boolean>("player.video_stats");
+const actionOnClick = useConfig<number>("player.action_onclick");
+
+const liveLatency = ref<string>();
+const videoStats = ref({
+	droppedFrames: 0,
+	playbackRate: 0,
+	bitrate: "0",
+	width: 0,
+	height: 0,
+	framerate: 0,
+});
+
+definePropertyHook(props.instance.component, "props", {
+	value: (v: Twitch.MediaPlayerComponent["props"]) => {
+		if (v.mediaPlayerInstance) {
```

Comment:
Assign `v.mediaPlayerInstance` to a ref and use a watcher. You can see an example of this here:
https://github.com/SevenTV/Extension/blob/master/src/site/twitch.tv/modules/chat/ChatController.vue#L246

By doing this we are making vue handle diffing and avoid re-hooking the component many times. This also ensures that if we receive a different pointer, the hooks can be removed from the previous one.